React/Base/RCTModuleData.mm

```diff
@@ -330,15 +333,17 @@ - (void)gatherConstants
 
 - (dispatch_queue_t)methodQueue
 {
-  (void)[self instance];
-  RCTAssert(_methodQueue != nullptr, @"Module %@ has no methodQueue (instance: %@, bridge.valid: %d)",
-            self, _instance, _bridge.valid);
+  if (_bridge.valid) {
+    id instance = self.instance;
+    RCTAssert(_methodQueue != nullptr, @"Module %@ has no methodQueue (instance: %@)",
+              self, instance);
+  }
   return _methodQueue;
 }
 
 - (void)invalidate
 {
-  _methodQueue = nil;
```

Comment:
Is there any reliable way to use `_methodQueue` as an indication that the module is "dead" instead of adding a new `_valid` ivar? No big deal, but would be better to re-use existing ivars if possible.

Reply:
The guarantee that `_valid` is always set on the JS thread is one I would like to keep. We can't guarantee that for `_methodQueue` since a native module could theoretically change its method queue from another thread.

Reply:
Oh, it seems that `_methodQueue` cannot be mutated by the native module's `_instance`, so we could use it here.